validphys2/src/validphys/photon/compute.py

```diff
@@ -276,13 +321,78 @@ def alpha_em_fixed_flavor(self, q, alpha_ref, qref, nf):
         alpha_em at NLO : float
             target value of a
         """
+        alpha_ref = couplings_ref[1]
         lmu = 2 * np.log(q / qref)
-        den = 1.0 + self.beta0[nf] * alpha_ref * lmu
+        den = 1.0 + self.betas_qed[nf][0] * alpha_ref * lmu
         alpha_LO = alpha_ref / den
-        alpha_NLO = alpha_LO * (1 - self.b1[nf] * alpha_LO * np.log(den))
-        return alpha_NLO
+        alpha_NLO = alpha_LO * (1 - self.betas_qed[nf][1] * alpha_LO * np.log(den))
+        return [couplings_ref[0], alpha_NLO]
+
+    def couplings_fixed_flavor_exa(self, q, couplings_ref, qref, nf):
+        """
+        Compute numerically the running running for nf fixed.
+        The RGEs for the two couplings are coupled by the mixed terms,
+        so must be solved together starting from the same qref.
+
+        Parameters
+        ----------
+        q : float
+            target scale
+        alph_aem_ref : float
+            reference value of alpha_em
+        qref: float
+            reference scale
+        nf: int
+            number of flavors
 
-    def set_thresholds_alpha_em(self):
+        Returns
+        -------
+        alpha_em at NLO : float
+            target value of a
+        """
+        u = 2 * np.log(q / qref)
+
+        # integration kernel
+        def rge(_t, alpha, beta_qcd_vec, beta_qcd_mix, beta_qed_vec, beta_qed_mix):
+            "RGEs for the couplings. See Eqs. (5-6) of arXiv:hep-ph/9803211"
+            rge_qcd = (
+                -(alpha[0] ** 2)
+                * beta_qcd_vec[0]
+                * (
+                    1
+                    + np.sum([alpha[0] ** (k + 1) * b for k, b in enumerate(beta_qcd_vec[1:])])
+                    + alpha[1] * beta_qcd_mix
+                )
+            )
+            rge_qed = (
+                -(alpha[1] ** 2)
+                * beta_qed_vec[0]
+                * (
+                    1
+                    + np.sum([alpha[1] ** (k + 1) * b for k, b in enumerate(beta_qed_vec[1:])])
+                    + alpha[0] * beta_qed_mix
+                )
+            )
+            res = np.array([rge_qcd, rge_qed])
+            return res
```

Comment:
Can this integration kernel be moved outside from the class since it doesn't use anything from `self.`?